ADDONS.md

```diff
@@ -301,6 +301,66 @@ module RubyLsp
 end
 ```
 
+### Registering for file update events
+
+By default, the Ruby LSP listens for changes to files ending in `.rb` to continuously update its index when Ruby source
+code is modified. If your your addon uses a tool that is configured through a file (like RuboCop and its `.rubocop.yml`)
+you can register for changes to these files and react when the configuration changes.
+
+**Note**: you will receive events from `ruby-lsp` and other addons as well, in addition to your own registered ones.
+
+
+```ruby
+module RubyLsp
+  module MyGem
+    class Addon < ::RubyLsp::Addon
+      def activate(global_state, message_queue)
+        register_additional_file_watchers(global_state, message_queue)
+      end
+
+      def deactivate; end
+
+      def register_additional_file_watchers(global_state, message_queue)
+      # Clients are not required to implement this capability
```

Comment:
Let's fix the indentation here.

Reply:
Good catch. Would you be open to add [rubocop-md](https://github.com/rubocop/rubocop-md) in a separate PR?

Reply:
Sure, I haven't tried it before but it looks useful.